src/main/java/org/tarantool/MsgPackLite.java

```diff
@@ -86,13 +133,13 @@ public void pack(Object item, OutputStream os) throws IOException {
         if (item == null) {
             out.write(MP_NULL);
         } else if (item instanceof Boolean) {
-            out.write(((Boolean) item).booleanValue() ? MP_TRUE : MP_FALSE);
+            out.write(((boolean) item) ? MP_TRUE : MP_FALSE);
```

Comment:
here, we have unnecessary type unboxing using `Boolean.booleanValue()` that is highlighted by the code inspection. And, yes, it can be more concise.

Reply:
I don't see a difference between `booleanValue()` and cast to `boolean`, to be honest, however okay: it is to eliminate a warning (I mean the highlight).

Is not it would be more logical to cast to `Boolean` (boxed) type?

Reply:
> Is not it would be more logical to cast to Boolean (boxed) type?

Looks redundant. 

Java ternary [operator](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.25) evaluates the first expression to `boolean` and decide either branch should be evaluated:

> ... At run time, the first operand expression of the conditional expression is evaluated first. If necessary, **unboxing** conversion is performed on the result.

> The resulting `boolean` value is then used to choose either the second or the third operand expression ...

Reply:
So, back to the change itself:

```diff
-            out.write(((Boolean) item).booleanValue() ? MP_TRUE : MP_FALSE);
+            out.write(((boolean) item) ? MP_TRUE : MP_FALSE);
```

We replaced cast to `Boolean` and unboxing with just unboxing. Okay so.